### PR TITLE
[MRG] Adding robust scaler preprocessor

### PR DESCRIPTION
**Pull request by hubertjb**
Robust scaler to be used in the USleep example (#282).

### CODE REVIEW

File: braindecode/preprocessing/preprocess.py [hunk 1]
```diff
@@ -274,23 +275,53 @@ def zscore(data):
...
+
+    .. note::
+        If this function is supposed to preprocess continuous data, it should be
+        given to raw.apply_function().
+    """
+    scaled = data - np.median(data, keepdims=True, axis=-1)
+    q75, q25 = np.percentile(scaled, [75, 25], keepdims=True, axis=-1)
+    scaled = scaled / (q75 - q25)
+    if hasattr(data, '_data'):
+        data._data = scaled
```
Comment (agramfort) [inline braindecode/preprocessing/preprocess.py:303]: I don’t like that. Docsting says data is ndarray that will be modified inplace but inplace occurs only for Evoked and Epochs (not explained from docstring) and assume hack a private attribute only present if preload = True. You should be using apply_function here.

Comment (hubertjb) [reply braindecode/preprocessing/preprocess.py]: It was an artifact of the other preprocessors - I agree with you. I removed them with the last commit.

File: braindecode/preprocessing/preprocess.py [hunk 2]
```diff
@@ -252,54 +252,66 @@ def zscore(data):
...
+    Returns
+    -------
+    np.ndarray :
+        Normalized continuous or windowed data of shape (n_channels, n_times)
+        or (n_windows, n_channels, n_times).
+    """
+    scaled = data - np.median(data, keepdims=True, axis=-1)
+    q75, q25 = np.percentile(scaled, [75, 25], keepdims=True, axis=-1)
+    scaled = scaled / (q75 - q25)
+    return scaled
```
Comment (agramfort) [inline braindecode/preprocessing/preprocess.py:294]: this is not operating in-place no?

**Comment by gemeinl** [comment]
Is there a reason you don't just use `sklearn.preprocessing.robust_scale`?
Like so:
```
from sklearn.preprocessing import robust_scale
from braindecode.preprocessing import Preprocessor, preprocess
preprocess(
    dataset,
    preprocessors=[Preprocessor(robust_scale, chanel_wise=True)],
)
```
Should give you exactly the same results.

**Comment by gemeinl** [comment]
Also `zscore` could potentially be removed and replaced by `sklearn.preprocessing.scale` if I am not mistaken.

**Comment by hubertjb** [comment]
> Also `zscore` could potentially be removed and replaced by `sklearn.preprocessing.scale` if I am not mistaken.

Ah you're right @gemeinl, I didn't know sklearn already had an implementation... I guess then we should remove both `robust_scale` and `zscore` and replace them with sklearn's  `robust_scale` and `scale`. Should we also remove braindecode's `scale` and recommend the direct use of `np.multiply`?

I'll close this PR and open a new one to remove these functions and their tests, what do you think? @agramfort @gemeinl

**Comment by robintibor** [comment]
sounds good, removing code always makes the world a bit more beautiful :D

**Comment by agramfort** [comment]
+1000

**Comment by hubertjb** [comment]
By the way, do you prefer we remove them entirely right now, or we start by deprecating them in 0.5.2?

**Comment by agramfort** [comment]
>
> By the way, do you prefer we remove them entirely right now, or we start
> by deprecating them in 0.5.2?
>

theory says:
-  any released public code should be deprecated
- you don't deprecate with minor versions so you can only have deprecation
if next release is 0.6 (which we should do)

**Pull request closed without merging**